pkg/reconciler/base.go

```diff
@@ -56,46 +56,48 @@ type GenericDeploymentReconciler struct {
 	// objects listers
 	Lister func(namespace string) appslistersv1.DeploymentNamespaceLister
 
-	*GenericRBACReconciler
+	*GenericRBACReconciler[T]
 }
 
 // GenericServiceReconciler contains interfaces shared across Service reconcilers.
-type GenericServiceReconciler struct {
+type GenericServiceReconciler[T kmeta.OwnerRefable] struct {
 	// URI resolver for sinks
 	SinkResolver *resolver.URIResolver
 	// API clients
 	Client func(namespace string) servingclientv1.ServiceInterface
 	// objects listers
 	Lister func(namespace string) servinglistersv1.ServiceNamespaceLister
 
-	*GenericRBACReconciler
+	*GenericRBACReconciler[T]
 }
 
 // GenericRBACReconciler reconciles RBAC objects for components adapters.
-type GenericRBACReconciler struct {
+type GenericRBACReconciler[T kmeta.OwnerRefable] struct {
 	// API clients
 	SAClient func(namespace string) coreclientv1.ServiceAccountInterface
 	RBClient func(namespace string) rbacclientv1.RoleBindingInterface
 	// objects listers
-	SALister func(namespace string) corelistersv1.ServiceAccountNamespaceLister
-	RBLister func(namespace string) rbaclistersv1.RoleBindingNamespaceLister
+	SALister     func(namespace string) corelistersv1.ServiceAccountNamespaceLister
+	RBLister     func(namespace string) rbaclistersv1.RoleBindingNamespaceLister
+	OwnersLister ListerGetter[T]
 }
```

Comment:
The most important part of the PR is here.

The typed listers that used to be embedded in each component's `Reconciler` are now moved to the `GenericRBACReconciler` as a generic lister type.